src/fprime_gds/common/testing_fw/pytest_integration.py

```diff
@@ -38,7 +39,19 @@ def pytest_addoption(parser):
         # Reduce flags to only the long option (i.e. --something) form
         flags = [flag for flag in flags if flag.startswith("--")]
         parser.addoption(*flags, **specifiers)
-
+        
+    # Add an option to specify JUnit XML report file
+    parser.addoption(
+        "--junit-xml-file",
+        action="store",
+        default="report.xml",
+        help="File to store JUnit XML report. [default: %(default)s]",
+    )
+
+def pytest_configure(config):
+    """ Create a JUnit XML report file to capture the test result """
```

Comment:
Could we start this docstring by saying this is a pytest hook ?
https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_configure

Other devs might benefit from knowing they can add stuff there if needed